### PR TITLE
[Mellanox] Enable dynamic buffer calculation after reloading minigraph for Mellanox default SKUs

### PR DESCRIPTION
**Pull request by stephenxs**
### Description of PR

Summary:
Support enabling dynamic buffer calculation after reloading minigraph on Mellanox platform

Signed-off-by: Stephen Sun <stephens@nvidia.com>

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
`config load_mingraph` is widely used to restore the configuration and state of the switch during the regression test.
However, after reloading minigraph the dynamic buffer calculation will be disabled, which means we can not run the regression test with the feature enabled.
So we need to enable dynamic buffer calculation after reloading minigraph.

#### How did you do it?
The idea is:
- Provide a script to enable the dynamic buffer calculation on the fly. This script should be deployed on the DUT during the deployment stage.
- If the dynamic buffer calculation is enabled before reloading minigraph, enable it after.
- Provide an argument to control this behavior.

#### How did you verify/test it?
Run the regression test.

### CODE REVIEW

**Comment by stephenxs** [comment]
All fixed.

File: tests/common/helpers/enable-dynamic-buffer.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+# Connect to the CONFIG_DB
+db_kwargs = {}
+config_db = ConfigDBConnector(**db_kwargs)
+config_db.db_connect('CONFIG_DB')
+
+# Don't enable dynamic buffer calculation if it is a MSFT SKU
+metadata = config_db.get_entry('DEVICE_METADATA', 'localhost')
+if 'ACS-MSN' not in metadata['hwsku']:
+    print("Don't enable dynamic buffer calculation for MSFT SKUs")
```
Comment (liat-grozovik) [inline tests/common/helpers/enable-dynamic-buffer.py:18]: please change to non-default SKUs.
default SKUs should have it enabled, non default can keep it disabled.

File: tests/common/helpers/enable-dynamic-buffer.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
```
Comment (liat-grozovik) [inline tests/common/helpers/enable-dynamic-buffer.py:1]: Can you please check if each part can be a function and one helper to call each in the right order? i think the code will be more readable in that way.

Comment (stephenxs) [reply tests/common/helpers/enable-dynamic-buffer.py:1]: Yes, we can. Will fix it.

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**